rand_distr/tests/cdf.rs

```diff
@@ -159,6 +158,377 @@ fn normal() {
+    }
+
+    let parameters = [
+        (0.0, 1.0),
+        (0.0, 0.1),
+        (1.0, 10.0),
+        (1.0, 100.0),
+        (-1.0, 0.00001),
+        (-1.0, 0.0000001),
+    ];
+
+    for (seed, (median, scale)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Cauchy::new(median, scale).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, median, scale));
+    }
+}
+
+#[test]
+fn uniform() {
+    fn cdf(x: f64, a: f64, b: f64) -> f64 {
+        if x < a {
+            0.0
+        } else if x < b {
+            (x - a) / (b - a)
+        } else {
+            1.0
+        }
+    }
+
+    let parameters = [(0.0, 1.0), (-1.0, 1.0), (0.0, 100.0), (-100.0, 100.0)];
+
+    for (seed, (a, b)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Uniform::new(a, b).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, a, b));
+    }
+}
+
+#[test]
+fn log_normal() {
+    fn cdf(x: f64, mean: f64, std_dev: f64) -> f64 {
+        if x <= 0.0 {
+            0.0
+        } else if x.is_infinite() {
+            1.0
+        } else {
+            0.5 * ((mean - x.ln()) / (std_dev * f64::consts::SQRT_2)).erfc()
+        }
+    }
+
+    let parameters = [
+        (0.0, 1.0),
+        (0.0, 0.1),
+        (0.5, 0.7),
+        (1.0, 10.0),
+        (1.0, 100.0),
+    ];
+
+    println!("{}", cdf(5.0, 2.0, 1.5));
+
+    for (seed, (mean, std_dev)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::LogNormal::new(mean, std_dev).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, mean, std_dev));
+    }
+}
+
+#[test]
+fn pareto() {
+    fn cdf(x: f64, scale: f64, alpha: f64) -> f64 {
+        if x <= scale {
+            0.0
+        } else {
+            1.0 - (scale / x).powf(alpha)
+        }
+    }
+
+    let parameters = [
+        (1.0, 1.0),
+        (1.0, 0.1),
+        (1.0, 10.0),
+        (1.0, 100.0),
+        (0.1, 1.0),
+        (10.0, 1.0),
+        (100.0, 1.0),
+    ];
+
+    for (seed, (scale, alpha)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Pareto::new(scale, alpha).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, scale, alpha));
+    }
+}
+
+#[test]
+fn exp() {
+    fn cdf(x: f64, lambda: f64) -> f64 {
+        1.0 - (-lambda * x).exp()
+    }
+
+    let parameters = [0.5, 1.0, 7.5, 32.0, 100.0];
+
+    for (seed, lambda) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Exp::new(lambda).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, lambda));
+    }
+}
+
+#[test]
+fn weibull() {
+    fn cdf(x: f64, lambda: f64, k: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+
+        1.0 - (-(x / lambda).powf(k)).exp()
+    }
+
+    let parameters = [
+        (0.5, 1.0),
+        (1.0, 1.0),
+        (10.0, 0.1),
+        (0.1, 10.0),
+        (15.0, 20.0),
+        // (1000.0, 0.001), // Fail case
```

Comment:
This case does not really make sense because of the small `k` (the expected values is bigger than the largest finite f64), I would use `(1000, 0.01)` instead which passes.